rfc/kubernetes-ui-enhancements/kubernetes-ui-enhancements.md

```diff
@@ -0,0 +1,260 @@
+
+Vertical Pod Autoscalers can be used to provide suggested values for a pod’s CPU and memory requests. VPAs can also automatically apply these recommended values. Similar to HPAs, VPAs can target workloads. there are enough conceptual similarities to extend the aforementioned HPA UI patterns to VPAs and display them in Spinnaker.
+
+### Persistent Volumes
+
+#### Current Functionality
+
+Although Spinnaker supports the deployment of Persistent Volume Claims, both Persistent Volumes and their associated claims are not rendered in the UI.
+
+#### Non-spinnaker Tooling
+
+Other Kubernetes UIs render Persistent Volume Claims and display the association to the underlying volume. This allows users to easily discern the status, access mode, and storage class of a given PVC.
+
+The [Kubernetes Dashboard tool](https://github.com/kubernetes/dashboard) also offers an extensive breakdown of the properties of the underlying volume. A PVC’s backing volume can be examined to determine its reclamation policy, source disk, and storage capacity.
+
+Finally, pods with associated Persistent Volume Claims display a link to the details of the PVC.
+
+#### Proposal
+
+The current UI offers no guidance on the relationship between a Pod, Persistent Volume Claim, and backing volume. The absence of PVCs, underlying volumes, and PVC pod links presents the opportunity to improve users understanding and enhance their ability to debug storage objects from within Spinnaker.
+
+The following set of changes are advised:
+
+- Render Persistent Volume Claims in the infrastructure view of Spinnaker’s UI
+  - Expose PVC specific metadata covering the access mode, storage class, and a reference to the underlying volume.
+  - Introduce a “Storage” infrastructure tab for rendering cluster level objects such as PVCs.
+  - Link to the underlying Persistent Volume for each PVC.
+- Render Persistent Volumes under the new Storage tab
+  - Expose Persistent Volume specific metadata on the reclamation policy, source disk, and storage capacity.
+  - Link back to the underling PVC.
+- Update the Pod UI to link to the associated PVC
+
+![](storage-mock.png)
+*The Spinnaker UI with a proposed Storage tab*
+
+![](pvc-mock.png)
+*A PVC listing mockup grouped by account and namespace.*
+
+### Config Maps
+
+#### Current Functionality
+
+Similar to its handling of Persistent Volume Claims, Spinnaker supports deploying Config Maps but does not render them in the UI or surface information on Config Map consumers. Other Kubernetes tools often feature a dedicated UI category for reviewing a cluster’s Config Maps.
+
+#### Proposal
+
+Similar to the effort to expose PVCs under a dedicated Storage tab, the UI components can be extended to surface Config Maps. These Config Maps will live in an additional UI tab for Configuration. A Config Maps’ stored data, a YAML viewer, and the following Metadata will be displayed:
+
+- Name
+- Namespace
+- Labels
+- Annotations
+- Created timestamp
+- Cluster
+
```

Comment:
Some kind of connection between deployments (or other objects) and the configmaps they use would be helpful.  And if there's enough information available to indicate that a configmap isn't being used at all, that would be huge.  Even if cleanup is left as an exercise to the reader, determining which configmaps aren't used is a bit painful at the moment.

Reply:
Aaah, I see you've addressed this at least partially below: https://github.com/spinnaker/governance/pull/94/files#diff-2952773823cfaff034e18ebc1ed38686R248.

Reply:
Yeah, if kubernetes had native GC, I wouldn't be so excited for spinnaker to help :)